op-chain-ops/cmd/op-upgrade/main.go

```diff
@@ -116,23 +126,37 @@ func entrypoint(ctx *cli.Context) error {
 		if err != nil {
 			return fmt.Errorf("cannot create RPC clients: %w", err)
 		}
+		// The L1Client is required
+		if clients.L1Client == nil {
+			return errors.New("Cannot create L1 client")
+		}
 
 		l1ChainID, err := clients.L1Client.ChainID(ctx.Context)
 		if err != nil {
 			return fmt.Errorf("cannot fetch L1 chain ID: %w", err)
 		}
-		l2ChainID, err := clients.L2Client.ChainID(ctx.Context)
-		if err != nil {
-			return fmt.Errorf("cannot fetch L2 chain ID: %w", err)
+
+		l2ChainID := chainConfig.ChainID
+
+		// The L2Client is not required, but double check the chain id matches if possible
+		if clients.L2Client != nil {
+			remoteL2ChainID, err := clients.L2Client.ChainID(ctx.Context)
+			if err != nil {
+				return fmt.Errorf("cannot fetch L2 chain ID: %w", err)
+			}
+			if remoteL2ChainID.Uint64() != l2ChainID {
+				return fmt.Errorf("Mismatched chain IDs: %d != %d", remoteL2ChainID.Uint64(), l2ChainID)
+			}
 		}
+
 		log.Info(chainConfig.Name, "l1-chain-id", l1ChainID, "l2-chain-id", l2ChainID)
 
 		log.Info("Detecting on chain contracts")
 		// Tracking the individual addresses can be deprecated once the system is upgraded
 		// to the new contracts where the system config has a reference to each address.
-		addresses, ok := superchain.Addresses[l2ChainID.Uint64()]
+		addresses, ok := superchain.Addresses[chainConfig.ChainID]
 		if !ok {
-			return fmt.Errorf("no addresses for chain ID %d", l2ChainID.Uint64())
+			return fmt.Errorf("no addresses for chain ID %d", l2ChainID)
```

Comment:
It's a bit confusing to use a different variable for the chain ID used to look up the addresses from what the log messages uses.  Both would probably be better as `l2ChainID` or just inline `l2ChainID` everywhere and always used `chainConfig.ChainID`.

Reply:
You are right that `l2ChainID` as a variable is probably not necessary